orientdb/src/main/java/org/hibernate/datastore/ogm/orientdb/dialect/impl/OrientDBTupleSnapshot.java

```diff
@@ -10,53 +10,137 @@
+		log.debugf( "2.associatedEntityKeyMetadata: %s", associatedEntityKeyMetadata );
+		log.debugf( "2.rolesByColumn: %s", rolesByColumn );
 	}
 
 	@Override
 	public Object get(String targetColumnName) {
-		LOG.info( "targetColumnName: " + targetColumnName );
-		return dbNameValueMap.get( targetColumnName );
+		log.debugf( "targetColumnName: %s", targetColumnName );
+		Object value = null;
+		if ( targetColumnName.equals( OrientDBConstant.SYSTEM_VERSION ) ) {
+			if ( dbNameValueMap.containsKey( OrientDBConstant.SYSTEM_VERSION ) ) {
+				value = dbNameValueMap.get( OrientDBConstant.SYSTEM_VERSION );
+			}
+			else {
+				value = Integer.valueOf( 0 );
+			}
+			log.debugf( "targetColumnName: %s, value: %d", targetColumnName, value );
+		}
+		else if ( targetColumnName.equals( "version" ) ) {
+			if ( dbNameValueMap.containsKey( OrientDBConstant.SYSTEM_VERSION ) ) {
+				value = dbNameValueMap.get( OrientDBConstant.SYSTEM_VERSION );
+			}
+			else {
+				value = Integer.valueOf( 0 );
+			}
+		}
+		/*
+		 * else if ( EntityKeyUtil.isEmbeddedColumn( targetColumnName ) ) { EmbeddedColumnInfo ec = new
+		 * EmbeddedColumnInfo( targetColumnName ); ODocument targetDocument = findEmbeddedClass( (ODocument)
+		 * dbNameValueMap.get( ec.getClassNames().get( 0 ) ), ec ); if (targetDocument!=null) { log.debugf(
+		 * "embedded column. class: %s ; property: %s", targetDocument.getClassName(), ec.getPropertyName() ); } else {
+		 * log.debugf( "embedded column.  class not found for property: %s", ec ); } value = targetDocument!=null ?
+		 * targetDocument.field( ec.getPropertyName() ) : null; log.debugf(
+		 * "targetColumnName: %s ; value: %s; class : %s", targetColumnName, value, ( value != null ? value.getClass() :
+		 * null ) ); }
+		 */
+		else if ( OrientDBConstant.MAPPING_FIELDS.containsKey( targetColumnName ) ) {
+			value = dbNameValueMap.get( OrientDBConstant.MAPPING_FIELDS.get( targetColumnName ) );
+		}
+		else {
+			value = dbNameValueMap.get( targetColumnName );
+			log.debugf( "targetColumnName: %s ; value: %s; class : %s", targetColumnName, value, ( value != null ? value.getClass() : null ) );
+		}
+		return value;
+	}
+
+	private ODocument findEmbeddedClass(ODocument doc, EmbeddedColumnInfo ec) {
+		ODocument targetDocument = doc;
+		if ( targetDocument == null ) {
+			return null;
+		}
+		if ( ec.getClassNames().size() > 1 ) {
+			for ( int i = 1; i < ec.getClassNames().size(); i++ ) {
+				String className = ec.getClassNames().get( i );
+				targetDocument = targetDocument.field( className );
+			}
+		}
+
+		return targetDocument;
+	}
+
+	private Map<String, Object> loadAssociatedEntity(AssociatedEntityKeyMetadata associatedEntityKeyMetadata, String targetColumnName) {
+		String mappedByName = AssociationUtil.getMappedByFieldName( associatedEntityKeyMetadata );
+		String inOrientDbField = "in_".concat( mappedByName );
+		log.debug( "mappedByName: " + mappedByName + "; inOrientDbField:" + inOrientDbField );
```

Comment:
it seems to me that there are to many debug messages, but, at least, use `log.debugf`